pkg/instancetype/instancetype.go

```diff
@@ -655,17 +656,53 @@ func (m *methods) inferDefaultsFromDataVolumeSourceRef(sourceRef *v1beta1.DataVo
 	return "", "", fmt.Errorf("unable to infer defaults from DataVolumeSourceRef as Kind %s is not supported", sourceRef.Kind)
 }
 
+func checkResourceRequestConflicts(field *k8sfield.Path, vmiSpec *virtv1.VirtualMachineInstanceSpec, instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec) Conflicts {
+	if _, hasRequestsCPU := vmiSpec.Domain.Resources.Requests[k8sv1.ResourceCPU]; hasRequestsCPU {
+		return Conflicts{field.Child("domain", "resources", "requests", string(k8sv1.ResourceCPU))}
+	}
+
+	if _, hasLimitsCPU := vmiSpec.Domain.Resources.Limits[k8sv1.ResourceCPU]; hasLimitsCPU {
+		return Conflicts{field.Child("domain", "resources", "limits", string(k8sv1.ResourceCPU))}
+	}
+
+	if _, hasRequestsMemory := vmiSpec.Domain.Resources.Requests[k8sv1.ResourceMemory]; hasRequestsMemory {
+		return Conflicts{field.Child("domain", "resources", "requests", string(k8sv1.ResourceMemory))}
+	}
+
+	if _, hasLimitsMemory := vmiSpec.Domain.Resources.Limits[k8sv1.ResourceMemory]; hasLimitsMemory {
+		return Conflicts{field.Child("domain", "resources", "limits", string(k8sv1.ResourceMemory))}
+	}
+
+	return nil
+}
+
+func applyResources(field *k8sfield.Path, instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec, preferenceSpec *instancetypev1alpha2.VirtualMachinePreferenceSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec) Conflicts {
+	if conflicts := checkResourceRequestConflicts(field, vmiSpec, instancetypeSpec); len(conflicts) > 0 {
+		return conflicts
+	}
+
+	// Set resources both memory and cpu
+	if instancetypeSpec.Resources != nil {
+		vmiSpec.Domain.Resources = *instancetypeSpec.Resources
+	}
+
+	return nil
+}
+
 func applyCpu(field *k8sfield.Path, instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec, preferenceSpec *instancetypev1alpha2.VirtualMachinePreferenceSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec) Conflicts {
 	if vmiSpec.Domain.CPU != nil {
 		return Conflicts{field.Child("domain", "cpu")}
 	}
 
-	if _, hasCPURequests := vmiSpec.Domain.Resources.Requests[k8sv1.ResourceCPU]; hasCPURequests {
-		return Conflicts{field.Child("domain", "resources", "requests", string(k8sv1.ResourceCPU))}
+	if instancetypeSpec.CPU == nil {
+		return nil
 	}
 
-	if _, hasCPULimits := vmiSpec.Domain.Resources.Limits[k8sv1.ResourceCPU]; hasCPULimits {
-		return Conflicts{field.Child("domain", "resources", "limits", string(k8sv1.ResourceCPU))}
+	// Check if resources were set
+	if instancetypeSpec.Resources != nil {
+		if _, hasRequestedCPU := instancetypeSpec.Resources.Requests[k8sv1.ResourceCPU]; hasRequestedCPU {
+			return nil
+		}
```

Comment:
This should be stopped by the validation webhook right?

It's valid to check again here but what about `Limits` as well? 

TBH you might want to extract this `hasCPUResourceRequests` logic into a standalone function within the module.